### PR TITLE
Delay UpdateNodeInstanceConnection message on children connections and disconnections

### PR DESCRIPTION
**Pull request by MrZammler**
##### Summary

This PR adds a small delay (10 seconds) to `UpdateNodeInstanceConnection` messages caused by children connecting and disconnecting.

Instead of sending the message immediately after connection and disconnection, it will instead register the timestamp of either the connection or disconnection, and if there are multiple of them during the 10 second period, it will actually send the last (connection or disconnection) 10 seconds after it occurred.

So in brief, if multiple such messages occur during a 10 second window, it will throttle them and only send the last one, whereas if they occur outside of the 10 second window, it will send them at once.

##### Test Plan

Setup a parent/child relationship.

Start the parent, then start the child.
Observe in log the entries: `Queuing status update for node=.....`

Start/stop the child during various intervals. Try to stop/start/stop the child during the 10 second window. A `Queuing status update for node=.....` should go out after 10 seconds of the last connect/disconnect event.

Try to start/stop/start etc the child with e.g. 15 second intervals. Messages should go out without any 10 second delay.

##### Additional Information

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

### CODE REVIEW

File: database/sqlite/sqlite_aclk_node.c [hunk 1]
```diff
@@ -130,3 +130,25 @@ void sql_build_node_info(struct aclk_database_worker_config *wc, struct aclk_dat
 
     return;
 }
+
+void register_update_node_instance_connection_msg(RRDHOST *host, int cmd)
```
Comment (thiagoftsm) [inline database/sqlite/sqlite_aclk_node.c:134]: Hello @MrZammler ,

I saw in next file that this function is only called inside `#ifdef ENABLE_ACLK`, so I will give a suggestion, instead to have the same `ifdef` inside the function, we could write the whole function inside an `#ifdef ENABLE_ACLK` block.  Or are you thinking to call the same function from more places?

Best regards!

Comment (MrZammler) [reply database/sqlite/sqlite_aclk_node.c:135]: True! Yes, moved the ifdef outside the function, thanks!

**Comment by underhood** [comment]
isn't 10 seconds a bit too excessive?

**Comment by MrZammler** [comment]
> isn't 10 seconds a bit too excessive? how did you decide this is the right value?

Well, in proper circumstances (where connects/disconnects are >10s apart) there won't be any delay to send the message. It will only kick in when connects/disconnects happen in less than 10s.

But we could do 5s as well if you think.

**Comment by underhood** [comment]
hmm i think I miss-understand the description here. better will check the code on how it works :D
I however have doubts that disconection/reconnection events in normal circumstances should be 10s appart.
How about parent that has 100 children. Wouldnt it be legitimate for it to send 100 update to online messages?

**Comment by MrZammler** [comment]
The debounce will be per host (it is logged in each one's worker thread). I believe it's ok for a parent to send 100 online messages.

The first online message will be sent immediately (just because wc is not active yet) and it's next offline again immediately. If a next online message comes <10s it will be delayed, if it comes >10 it will be sent again immediately.

Yes, the problem is when we send 100 per host very quickly. This is what was observed, in most cases only a few children of a parent behave like this, most of them were ok.

**Comment by MrZammler** [comment]
I think we could/should do less than 10s, maybe 5?

**Pull request closed without merging**